Main_project_Tests_TestUnit.rb

```diff
@@ -15,134 +15,187 @@ def setup
+	    issue1 = create_new_issue({type: '1'})
+	    issue2 = create_new_issue({type: '2'})
+	    issue3 = create_new_issue({type: '3'})
      	
- #     	actual_array_subjects = []
- #     	actual_array_subjects = @driver.find_elements(:css, '.subject').map {|element| element.text}
+     	actual_array_subjects = []
+     	actual_array_subjects = @driver.find_elements(:css, '.subject').map {|element| element.text}
+     	assert_equal(issue3.name, actual_array_subjects[0])
+     	assert_equal(issue2.name, actual_array_subjects[1])
+     	assert_equal(issue1.name, actual_array_subjects[2])
+     end
 
- #     	assert_equal(issue3.name, actual_array_subjects[0])
- #     	assert_equal(issue2.name, actual_array_subjects[1])
- #     	assert_equal(issue1.name, actual_array_subjects[2])
- #     end
-
-    # def test_make_random_action
-    # 	# make a random action: create or not a new bug issue
-    # end
 
     def test_open_project
     	user = register_user({})
     	project = create_new_project({})
     	open_project(project)
-    	
+
+    	expected_subject = project.name.to_s
+    	actual_subject = @driver.find_element(:css, 'h1').text
+    	assert_equal(actual_subject, expected_subject)
     end
 
-    # def test_open_issues_tab
-    # 	# open issues tab
-    # end
 
-    # def test_add_self_to_watchers
-    # open issues tab
-	# if there is a bug present then add self to watchers
-	# if no then create a new bug issue and then add self to watchers
-	# Verify that there is a bug issue and that current user is a watcher
+    def test_if_issues_tab_is_empty
+    	user = register_user({})
+     	project = create_new_project({})
+     	open_issues_tab(project)
+     	
+     	assert(@driver.find_element(:css, '.nodata').displayed?)
     end
 
+	def test_open_issues_tab
+		user = register_user({})
+ 	  	project = create_new_project({})
+ 	  	open_issues_tab(project)
+		
+		assert(@driver.find_element(:css, '.issues.selected').displayed?)
+	end
+
+	def test_add_self_to_watchers
+		user = register_user({})
+		project = create_new_project({})
+		issue = create_new_issue({})
+		add_self_to_watchers(project, issue)
 
+		self_watcher = user.first_name + ' ' + user.last_name
+		actual_watchers = []
+		actual_watchers = @driver.find_elements(:css, 'ul.watchers').map {|element| element.text}
+		assert_true(actual_watchers.include? self_watcher)
+	end
 
 
 
-end
\ No newline at end of file
+	# Create project, make a random action: create or not a new bug issue, open project page, open issues tab, if there is a bug present then add self to watchers if no then create a new bug issue and then add self to watchers
+	# Verify that there is a bug issue and that current user is a watcher
+
+	def test_create_issue_and_add_self_to_watchers
+		user = register_user({})
+		project = create_new_project({})
+		issue = create_or_not_create_new_issue(project)
+		if is_issues_tab_empty?(project) == true
+			issue = create_new_issue({})
+			add_self_to_watchers(project, issue)
```

Comment:
this line should be extracted from both ```if``` and ```else``` code blocks
and placed after ```if``` statement